Updated README to show basic usage

btkelly opened the pull request:
Resolves #11

stkent commented on README.md:
```diff
@@ -1 +1,119 @@
+
+# Usage
+
+The goal of Gandalf was to add this basic boiler plate code you any application quickly. You will need to add the following code to your application as well as host a json file on a publicly accessible server with the format included below.
```
"code you any" -> "code to any"

stkent replied:
"json" -> "JSON"

jsibbold replied:
json -> "JavaScript Object Notation"

stkent replied:
![s0ghjjh](https://cloud.githubusercontent.com/assets/6463980/12728475/023c0184-c8f0-11e5-86c2-6dbec240e2ef.gif)

stkent commented on README.md:
```diff
@@ -1 +1,119 @@
+    new Gandalf.Installer()
+            .setContext(this)
+            .setPackageName("com.my.package")
+            .setBootstrapUrl("http://www.example.com/bootstrap.json)
```
`.setBootstrapUrl("http://www.example.com/bootstrap.json)` -> `.setBootstrapUrl("http://www.example.com/bootstrap.json")`

stkent commented on README.md:
```diff
@@ -1 +1,119 @@
+    package="io.github.btkelly.gandalf.example">
+
+    <application
+        android:name=".CustomApplication"
```
If you're using this name here, perhaps also include it in the custom app class definition above?

stkent commented on README.md:
````diff
@@ -1 +1,119 @@
+</manifest>
+```
+
+#### Json File
````
"Json" -> "JSON"

stkent commented on README.md:
```diff
@@ -1 +1,119 @@
+
+#### Json File
+
+You must host a json file remotely and set the url of this file in the the Gandalf installer. The Json file must match the following format and use the Android `versionCode` not the `versionName` for version information.
```
"json" -> "JSON"

stkent replied:
"url" -> "URL"

stkent replied:
"Json" -> "JSON"

stkent commented on README.md:
```diff
@@ -1 +1,119 @@
+      "blocking": true
+    },
+    "optionalUpdate": {
+      "optionalVersion": "6",
```
Hmm, maybe better to use `optionalVersionCode` here to reinforce the requirement stated above?

btkelly replied:
wont fix

stkent commented on README.md:
```diff
@@ -1 +1,119 @@
+      "message": "A new version of the application is available, please click below to update to the latest version."
+    },
+    "requiredUpdate": {
+      "minimumVersion": "7",
```
As above.

btkelly replied:
wont fix

stkent commented on README.md:
````diff
@@ -1 +1,119 @@
+```
+
+
+That's all that's needed to get Gandalf up and running using the basic settings, if extending `GandalfActivity` doesn't work for you the `Gandalf` class can be used directly by calling `shallIPass(GandalfCallback callback)`. 
````
It might be good to be a little more explicit here:

> the `Gandalf` class can be used directly by calling `shallIPass(GandalfCallback callback)`

and mention that in this case, the user would need to write their own implementation of XX, YY and ZZ.

stkent commented on README.md:
```diff
@@ -1 +1,119 @@
+
+## License
+
+Copyright 2016 Bryan Kelly
```
Indent for nice formatting?

stkent replied:
Still open

btkelly replied:
Hmm wont fix for now, need to find a way to make this automatic in AS

stkent commented on README.md:
````diff
@@ -1 +1,124 @@
+Extend the Android [`Application`](http://developer.android.com/reference/android/app/Application.html) class and add the following to the `onCreate()`
+
+```java
+public class App extends Application {
````
Should use `CustomApplication` here to match the manifest, or use `App` in the manifest to match this.

btkelly replied:
Whoops, thought I did that

btkelly commented:
I addressed most of the comments but I don't want to change the JSON format as we discussed that with @dtrenz and we wanted them to be the same for each platform.

stkent commented:
Approved

stkent commented:
Approved
